docker/compose/fix-rustfmt.yaml

```diff
@@ -0,0 +1,102 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+version: '3.6'
+
+services:
+
+  fix-rustfmt-core:
+    build:
+      context: .
+      dockerfile: ../lint.dockerfile
```

Comment:
If running `rust fmt`, maybe we should rename to `fmt.dockerfile` or something generic if intended for multiple purposes.

Reply:
lint.dockerfile has all necessary rust components to run lint checks installed in it. We are reusing that file here and running explicitly cargo fmt for each sub-projects.

Reply:
There is a separate clippy.dockerfile alongside. I think clippy is what takes care of lint. Or are we not planning to use it? That is what confuses me.

Reply:
Clippy errors out for improper use of programming constructs, lint filters out styling errors. Together they perform lint checks.